### PR TITLE
Update draft-ietf-tls-external-psk-guidance.md

### PR DESCRIPTION
**Pull request by russhousley**
This deals with many comments from the AD Evaluation.  The comment on section 6.1 still needs to be resolved, and the NITS need to be addressed.

### CODE REVIEW

File: draft-ietf-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-ietf-tls-external-psk-guidance.md]:
```suggestion
PSKs allows TLS endpoints to authenticate the endpoint identities.  However,
```

File: draft-ietf-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-ietf-tls-external-psk-guidance.md:174]: I would recommend deleting this entire sentence, for two reasons: (1) the server identity is still leaked even if certificates are used (and ECH is not used), and (2) there may be other future forms of authentication that prohibit deniability. It seems reasonable to just say "PSK-based auth has these properties," and nothing more.

File: draft-ietf-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-ietf-tls-external-psk-guidance.md]: I'd revert this text too. @kaduk's comment on selfie doesn't seem to be in the scope of that attack, so IMO the previous text more accurately describes things.

File: draft-ietf-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-ietf-tls-external-psk-guidance.md]:
```suggestion
repeating that there are concerns related to the use of external PSKs regarding
proper identification of TLS 1.3 endpoints and additional risks when external 
PSKs are known to a group.
```

File: draft-ietf-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-ietf-tls-external-psk-guidance.md]:
```suggestion
in Transport Layer Security (TLS) 1.3 as defined in RFC 8446.
```

File: draft-ietf-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-ietf-tls-external-psk-guidance.md]:
```suggestion
provided by TLS 1.3 when external PSKs are used.
```

File: draft-ietf-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-ietf-tls-external-psk-guidance.md]:
```suggestion
in Transport Layer Security (TLS) 1.3 {{RFC8446}}. This guidance also
applies to Datagram TLS (DTLS) 1.3 {{I-D.ietf-tls-dtls13}} and
Compact TLS 1.3 {{I-D.ietf-tls-ctls}}. For readability, this document uses
the term TLS to refer to all such versions. External PSKs are symmetric 
secret keys provided to the TLS protocol implementation as external inputs. 
External PSKs are provisioned out-of-band. This document lists
```

**Review by chris-wood — CHANGES_REQUESTED** [review]
Thanks! I think we should revert some changes here, but perhaps @kaduk has thoughts on the comments? 

Once this lands, I'll prepare a PR to address Ben's nits.

File: draft-ietf-tls-external-psk-guidance.md
Comment (kaduk) [inline draft-ietf-tls-external-psk-guidance.md:51]: "SHOULD [use importers]" seems like it would push this one into the normative section.

File: draft-ietf-tls-external-psk-guidance.md
Comment (kaduk) [reply draft-ietf-tls-external-psk-guidance.md]: I don't think that even strict use of SNI renders the class of attack "not possible", full stop.
But your point that this is tangential and we may not need to go into quite so much detail seems reasonable.

File: draft-ietf-tls-external-psk-guidance.md
Comment (kaduk) [inline draft-ietf-tls-external-psk-guidance.md]:
```suggestion
to connect them in unintended ways, as described below. Note that a partial mitigiation
against this class of attack is available: each group member includes the SNI extension {{RFC6066}}
and terminates the connection on mismatch between the presented SNI value and the
receiving member's known identity.  See {{Selfie}} for details.
```

File: draft-ietf-tls-external-psk-guidance.md
Comment (kaduk) [inline draft-ietf-tls-external-psk-guidance.md:444]: Apparently I didn't find this noteworthy the first time around, but reading it again, "RECOMMENDED to always use a single fixed GUID" has pretty lousy privacy properties.  Even if we think this is the right tradeoff based on the functionality needs, maybe we should acknowledge that it has poor privacy properties.

**Review by kaduk — APPROVED** [review]
Chris's suggestions generally seem reasonable

File: draft-ietf-tls-external-psk-guidance.md
Comment (chris-wood) [reply draft-ietf-tls-external-psk-guidance.md:51]: Agreed.

File: draft-ietf-tls-external-psk-guidance.md
Comment (chris-wood) [reply draft-ietf-tls-external-psk-guidance.md:444]: Couldn't hurt! Good suggestion.

File: draft-ietf-tls-external-psk-guidance.md
Comment (russhousley) [reply draft-ietf-tls-external-psk-guidance.md:174]: I think you are suggesting:

When using external PSK authentication, these TLS 1.3 does not provide privacy protection ...

If that is what you mean, then I a comfortable.
If you mean something else, please provide words.

File: draft-ietf-tls-external-psk-guidance.md
Comment (russhousley) [reply draft-ietf-tls-external-psk-guidance.md]: I do not see how reverting addresses the original comment.

File: draft-ietf-tls-external-psk-guidance.md
Comment (russhousley) [reply draft-ietf-tls-external-psk-guidance.md:444]: s/It is RECOMMENDED that each/One solution is that/

File: draft-ietf-tls-external-psk-guidance.md
Comment (chris-wood) [reply draft-ietf-tls-external-psk-guidance.md:174]: Yeah, that's what I mean.

File: draft-ietf-tls-external-psk-guidance.md
Comment (chris-wood) [reply draft-ietf-tls-external-psk-guidance.md:444]: Yep, that'd work.

**Review by chris-wood — COMMENTED** [review]
LGTM pending the RECOMMENDATION change.

**Review by chris-wood — APPROVED** [review]

**Pull request merged**
